### PR TITLE
adds softmax_scale to flash attention

### PR DESCRIPTION
**Pull request by codestar12**
extends the option to scale softmax to flash attention in addition to triton flash attention

### CODE REVIEW

**Review by vchiley — APPROVED** [review]
lgtm

Hopefully you resolve the GPU count issue described offline

**Review by bcui19 — APPROVED** [review]
LGTM! Thanks for adding this!

File: examples/llm/src/models/layers/attention.py [hunk 1]
```diff
@@ -122,16 +122,18 @@ def __init__(self, cfg: DictConfig, device: Optional[str] = None):
...
         self.d_model = cfg.d_model
         self.n_heads = cfg.n_heads
 
-        if self.attn_qk_ln or self.clip_qkv:
+        if self.attn_qk_ln or self.clip_qkv or self.softmax_scale:
             self.W_qkv = nn.Linear(self.d_model,
                                    3 * self.d_model,
                                    bias=True,
                                    device=device)
             self.inner_attn = FlashAttention(attention_dropout=cfg.attn_pdrop,
```
[the remark this answers is not in the document]
Comment (vchiley) [reply examples/llm/src/models/layers/attention.py:134]: `1/srqt(d/n_heads)` is standard.
`1/(d/n_heads)` is the recommended muP mod.
Since its part of the model config, it'll get dumped into wandb if we ever need to check what we used.
